backend/api/firebase_auth.py

```
@@ -0,0 +1,64 @@
+    firebase_app = firebase_admin.initialize_app(cred)
+except ValueError:
+    # App already initialized
+    firebase_app = firebase_admin.get_app()
+
+class FirebaseAuthentication(authentication.BaseAuthentication):
+    """
+    Firebase Authentication for Django REST Framework
+    Clients should authenticate by passing the Firebase ID token in the
+    Authorization header, prepended with the string "Bearer ".
+    """
+    def authenticate(self, request):
```

Ruff (0.8.2)

Missing return type annotation for public function `authenticate`

(ANN201)